src/main/java/io/quarkus/gizmo2/impl/GizmoImpl.java

```diff
@@ -1,147 +1,53 @@
-        if (current == null) {
-            throw new IllegalStateException("No active Gizmo");
-        }
-        return current;
+    public Gizmo withOutput(final ClassOutput outputHandler) {
+        return new GizmoImpl(outputHandler);
     }
 
     public ClassDesc class_(final ClassDesc desc, final Consumer<ClassCreator> builder) {
-        return defaultOutput.class_(desc, builder);
-    }
-
-    public ClassDesc interface_(final ClassDesc desc, final Consumer<InterfaceCreator> builder) {
-        return defaultOutput.interface_(desc, builder);
-    }
-
-    <T> void do_(T att, Consumer<T> task) {
-        GizmoImpl old = current.get();
-        if (old == this) {
-            task.accept(att);
-        } else {
-            current.set(this);
-            try {
-                task.accept(att);
-            } finally {
-                if (old == null) {
-                    current.remove();
-                } else {
-                    current.set(old);
-                }
-            }
+        if (!desc.isClassOrInterface()) {
+            throw new IllegalArgumentException("Descriptor must describe a valid class");
         }
-    }
-
-    private final ConcurrentHashMap<FieldDesc, StaticFieldVarImpl> staticFields = new ConcurrentHashMap<>();
-
-    public NullConstant nullConstant(ClassDesc type) {
-        return new NullConstant(type);
-    }
-
-    public ClassConstant classConstant(ClassDesc value) {
-        return new ClassConstant(value);
-    }
-
-    public StringConstant stringConstant(String value) {
-        return new StringConstant(value);
-    }
-
-    public ByteConstant byteConstant(Byte value) {
-        return new ByteConstant(value);
-    }
-
-    public ShortConstant shortConstant(Short value) {
-        return new ShortConstant(value);
-    }
-
-    public CharConstant charConstant(Character value) {
-        return new CharConstant(value);
-    }
-
-    public IntConstant intConstant(Integer value) {
-        return new IntConstant(value);
-    }
-
-    public LongConstant longConstant(Long value) {
-        return new LongConstant(value);
-    }
-
-    public FloatConstant floatConstant(float value) {
-        return new FloatConstant(value);
-    }
-
-    public DoubleConstant doubleConstant(double value) {
-        return new DoubleConstant(value);
-    }
-
-    public StaticFieldVar staticField(final FieldDesc desc) {
-        return staticFields.computeIfAbsent(desc, StaticFieldVarImpl::new);
-    }
-
-    public StaticFinalFieldConstant staticFinalFieldConstant(final FieldDesc desc) {
-        return new StaticFinalFieldConstant(desc);
-    }
 
-    public ClassOutput classOutput(final BiConsumer<ClassDesc, byte[]> outputHandler) {
-        return new ClassOutputImpl(this, outputHandler);
+        ClassFile cf = ClassFile.of(ClassFile.StackMapsOption.GENERATE_STACK_MAPS);
+        byte[] bytes = cf.build(desc, zb -> {
+            zb.withVersion(ClassFile.JAVA_17_VERSION, 0);
```

Comment:
Do we need this? It's gonna be overwritten by `preAccept`. If we do need this, maybe there should be a comment that the value is not gonna survive?